aries_vcx/docs/guidelines.md

```diff
@@ -0,0 +1,71 @@
+# Guidelines for implementing protocol state machines
+
+## No 'Sent' states
+Do not create states to track whether a reply has been sent to counterparty. It is responsibility of
+`aries-vcx` consumers to keep track of whether response message has been sent to counterparty. From state
+machine perspective, creation of a protocol message implies a reply to that message can be processed.
+
+Example: Do not create states such as `PresentationSent` (in context of `presentation-protocol` in role of prover).
+Instead, only create state `PresentationPrepared` which enables `aries-vcx` consumers to get the presentation message
+and send it to the counterparty. Subsequently, it should be possible to further drive state machine in  `PresentationPrepared` 
+using `presentation-ack` message received from verifier. 
+
+## Separate final states
+While Aries RFCs commonly refer to single final state to convey nothing more can happen after exchange
```

Comment:
I would say "sometimes" instead of "commonly", which may be too strong. For example. did exchange protocol has 2 terminal states.